### PR TITLE
[15 min fix] Added support for the --no-seed argument to bin/e2e command 

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update
- [x] DX

## Description

Adds support for running end to end (E2E) tests that do not require seeded data. If a developer wants to run E2E tests that do not use seeded data, they add the `--no-seed` argument, e.g. `bin/e2e --no-seed`.  There are also a npm script to handle this, `yarn e2e:noseed`.

## Related Tickets & Documents

#14285

## QA Instructions, Screenshots, Recordings

1. Open a terminal in the root folder of the Forem repository and run `yarn e2e:noseed`.
2. Once the checks are verified, it will output `"Running E2E tests without seed data"` before starting Cypress.

```bash
...
[84] ./app/javascript/packs/dashboardTagsDisableUnchangedButtons.js 754 bytes {26} [built]
    + 112 hidden modules

Running E2E tests without seed data

cypress-rails configuration:
============================
 CYPRESS_RAILS_DIR....................."/Users/nickytonline/dev/forem"
 CYPRESS_RAILS_HOST...................."127.0.0.1"
...
```

3. The Cypress test runner will launch with tests that required the database to not be seeded.

image

### UI accessibility concerns?

N/A Testing infrastructure

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: N/A Testing infrastructure
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

A character from Adventure Time saying "Nice!"

### CODE REVIEW

File: bin/e2e [hunk 1]
```diff
@@ -29,4 +29,10 @@ fi
 printf "Starting the test runner...\n\n"
 
 bundle exec rails assets:precompile;
-CYPRESS_RAILS_CYPRESS_OPTS="--config-file cypress.dev.json" RAILS_ENV=test E2E=true bundle exec rake cypress:open
+
+if [ $1 = "--no-seed" ]; then
```
Comment (nickytonline) [inline bin/e2e:33]: I'm explicitly checking `$1` because we only support one argument at the moment.

**Comment by djuber** [comment]
Works as advertised. Launched the cypress runner with the non-seeded flows focused (and the one trivial test passes).

**Review by djuber — APPROVED** [review]

**Review by msarit — APPROVED** [review]
Works like a charm! 🚀

File: bin/e2e (hunk 1 above)
Comment (zeotuan) [reply bin/e2e:33]: Hi i have been getting ```bin/e2e: line 33: [: =: unary operator expected``` when running e2e test and not all the test is shown 
image

**Pull request merged**